src/Dapr.Extensions.Configuration/DaprSecretStoreConfigurationProvider.cs

```diff
@@ -101,29 +101,27 @@ public DaprSecretStoreConfigurationProvider(string store, bool normalizeKey, IRe
         /// Creates a new instance of <see cref="DaprSecretStoreConfigurationProvider"/>.
         /// </summary>
         /// <param name="store">Dapr Secre Store name.</param>
-        /// <param name="normalizeKey">Replace "__" and any custom delimiters with the delimiter ":".</param>
-        /// <param name="customDelimiters">A collection of delimiters that will be replaced by ':' in the key of every secret.</param>
+        /// <param name="normalizeKey">Indicates whether any key delimiters should be replaced with the delimiter ":".</param>
+        /// <param name="keyDelimiters">A collection of delimiters that will be replaced by ':' in the key of every secret.</param>
         /// <param name="metadata">A collection of metadata key-value pairs that will be provided to the secret store. The valid metadata keys and values are determined by the type of secret store used.</param>
         /// <param name="client">Dapr client used to retrieve Secrets</param>
-        public DaprSecretStoreConfigurationProvider(string store, bool normalizeKey, IEnumerable<string> customDelimiters, IReadOnlyDictionary<string, string> metadata, DaprClient client)
+        public DaprSecretStoreConfigurationProvider(string store, bool normalizeKey, IEnumerable<string> keyDelimiters, IReadOnlyDictionary<string, string> metadata, DaprClient client)
         {
             ArgumentVerifier.ThrowIfNullOrEmpty(store, nameof(store));
             ArgumentVerifier.ThrowIfNull(client, nameof(client));
 
             this.store = store;
             this.normalizeKey = normalizeKey;
-            this.customDelimiters = customDelimiters;
+            this.keyDelimiters = keyDelimiters;
             this.metadata = metadata;
             this.client = client;
         }
 
         private string NormalizeKey(string key)
         {
-            key = key.Replace("__", ConfigurationPath.KeyDelimiter);
-
-            if (this.customDelimiters?.Any() == true)
+            if (this.keyDelimiters?.Any() == true)
```

Comment:
If you can change this to `IList` then you can get rid of `.Any()` and and use the `.Count` property